### PR TITLE
Note missing plugins support in mdxRs, refine

### PR DESCRIPTION
**Pull request by karlhorky**
### What?

1. Document incompatibility of Turbopack with MDX remark and rehype plugins
2. Improve casing and prose
3. Add link to `remark-gfm`

### Why?

It's confusing to users that plugins don't work with Turbopack (because `mdxRs` is required, which [does not support plugins](https://github.com/wooorm/mdxjs-rs#:~:text=This%20project%20does%20not%20yet%20support%20plugins.)):

- https://bsky.app/profile/matthamlin.bsky.social/post/3la2lzzuprj22
- https://github.com/vercel/next.js/issues/71819
- https://github.com/vercel/next.js/issues/71856

### How?

Edit [Remark and Rehype plugins](https://nextjs.org/docs/app/building-your-application/configuring/mdx#remark-and-rehype-plugins) section on docs page

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: ba517874036ef65a022b18dc6989a341a5148386

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by ijjk — APPROVED** [review]

**Comment by karlhorky** [comment]
Thanks for the review and merge!

File: docs/02-app/01-building-your-application/07-configuring/05-mdx.mdx [hunk 1]
```diff
@@ -643,10 +643,14 @@ const withMDX = createMDX({
...
 })
 
-// Wrap MDX and Next.js config with each other
+// Combine MDX and Next.js config
 export default withMDX(nextConfig)
 ```
 
+> **Good to know**:
+>
+> remark and rehype plugins cannot be used with [Turbopack](/docs/architecture/turbopack), due to lack of plugins support in [`mdxRs`](#using-the-rust-based-mdx-compiler-experimental)
```
Comment (timneutkens) [inline docs/02-app/01-building-your-application/07-configuring/05-mdx.mdx:652]: This message is wrong and the good to know [here](https://nextjs.org/docs/app/building-your-application/configuring/mdx#using-the-rust-based-mdx-compiler-experimental) is also outdated.

- Turbopack works with `@next/mdx`
- The reason you can't use `remark` / `rehype` plugins with Turbopack is that they are JavaScript functions and you can't pass JavaScript to Rust. Can link to this issue: https://github.com/vercel/next.js/issues/71819

Comment (karlhorky) [reply docs/02-app/01-building-your-application/07-configuring/05-mdx.mdx:652]: Oh sorry, wasn't aware of that detail! PR open:

- https://github.com/vercel/next.js/pull/72467

**Pull request merged**